ConsentViewController/Classes/SourcePointClient/SimpleClient.swift

```diff
@@ -95,28 +95,23 @@ class SimpleClient: HttpClient {
                 self?.logResponse(urlRequest, data)
 
                 if error != nil {
-                    var spError = GenericNetworkError(request: urlRequest, response: response as? HTTPURLResponse)
                     if let response = response as? HTTPURLResponse {
                         switch response.statusCode {
-                        case 400...499:
-                            spError = ResourceNotFoundError(request: urlRequest, response: response)
-
-                        case 500...599:
-                            spError = InternalServerError(request: urlRequest, response: response)
+                        case 408:
+                            handler(.failure(ConnectionTimeoutAPIError(apiCode: apiCode)))
 
                         default:
-                            spError = GenericNetworkError(request: urlRequest, response: response)
+                            handler(.failure(GenericNetworkError(request: urlRequest, response: response)))
                         }
                     }
-                    handler(.failure(spError))
                 } else {
                     handler(.success(data))
                 }
             }
         }.resume()
     }
 
-    func post(urlString: String, body: Data?, handler: @escaping ResponseHandler) {
+    func post(urlString: String, body: Data?, apiCode: NetworkCallErrorsCode.InvalidResponsAPICode?, handler: @escaping ResponseHandler) {
```

Comment:
Here you could use a default parameter so you don't have to check for null (and `?? .EMPTY`). Something like:
```swift
func post(urlString: String, body: Data?, apiCode: NetworkCallErrorsCode.InvalidResponsAPICode? = .EMPTY, handler: @escaping ResponseHandler) {
```

Reply:
![image](https://github.com/SourcePointUSA/ios-cmp-app/assets/17005391/8a706e10-aa0f-41b1-b511-5452bf822301)

Reply:
I can overload a method if you want

Reply:
@Nevazhnovu protocols don't accept default parameters, but their concrete implementations do.